flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/TableEnvHiveConnectorITCase.java

```diff
@@ -669,6 +669,18 @@ public void testNonExistingPartitionFolder() throws Exception {
 		}
 	}
 
+	@Test
+	public void testInsertPartitionWithStarSource() throws Exception {
+		hiveShell.execute("create table src (x int,y string)");
+		hiveShell.insertInto("default", "src").addRow(1, "a").commit();
+		hiveShell.execute("create table dest (x int) partitioned by (p1 int,p2 string)");
+		TableEnvironment tableEnv = getTableEnvWithHiveCatalog();
+		TableEnvUtil.execInsertSqlAndWaitResult(tableEnv,
+				"insert into dest partition (p1=1) select * from src");
+		List<Row> results = Lists.newArrayList(tableEnv.sqlQuery("select * from dest").execute().collect());
+		assertEquals("[1,1,a]", results.toString());
```

Comment:
Need to drop table `src` and `dest` after the test, because the catalog is reused across all test cases. And you can just use `tableEnv` to execute DDLs -- it's using the hive dialect